src/features.rs

```diff
@@ -238,29 +235,29 @@ impl DenoiseFeatures {
             r[i] *= (self.ex[i] / (1e-8 + self.ep[i])).sqrt();
         }
         crate::interp_band_gain(&mut rf[..], &r[..]);
-        for i in 0..FREQ_SIZE {
-            self.x[i] += rf[i] * self.p[i];
-        }
+        let rf: &mut [f32] = &mut rf;
+        self.x += &self.p * &*rf;
 
         let mut new_e = [0.0; NB_BANDS];
         crate::compute_band_corr(&mut new_e[..], &self.x, &self.x);
         for i in 0..NB_BANDS {
             r[i] = (self.ex[i] / (1e-8 + new_e[i])).sqrt();
         }
         crate::interp_band_gain(&mut rf[..], &r[..]);
-        for i in 0..FREQ_SIZE {
-            self.x[i] *= rf[i];
-        }
+        self.x *= &*rf;
     }
 
     pub(crate) fn apply_gain(&mut self, gain: &[f32; FREQ_SIZE]) {
-        for (x, g) in self.x.iter_mut().zip(gain) {
-            *x *= *g;
-        }
+        self.x *= gain as &[f32];
     }
 
     pub(crate) fn frame_synthesis(&mut self, out: &mut [f32]) {
-        self.fft.inverse(&mut self.x, &mut self.window_buf[..]);
+        self.window_buf.copy_from_slice(&self.x.real_ifft());
+        // Not too sure why this scaling factor is introduced
```

Comment:
This is a little scary; I'd like to understand it better. Could it be because we weren't correctly handling the complex/real length difference before?

Reply:
Yeah, I think you might be the best person to determine where it is. I suspect it's in some normalisation code that adds the factor and your `inverse` function happened to remove it again. Didn't find an obvious candidate when I had a look.

Reply:
Fair enough, I will try to figure this out over the next day or two.